### DISCUSSION TITLE
Create fetch with default options

### DISCUSSION THREAD

**Discussion by Gyurmatag** [nuxt/nuxt#17813] [Questions] [answered]
I want to make a base $fetch using ohmyfetch like this:
`const apiFetch = $fetch.create({ baseURL: 'www.base.com' })
`And then call it like this:
`apiFetch('/test')
`It's in the docs here: https://github.com/unjs/ohmyfetch#%EF%B8%8F-create-fetch-with-default-options

I cant use this is Nuxt 3. How should I do this? I want to create it inside a composable so I can use it everywhere without imports.

Have you got any ideas? 

Thank you!

**Comment by Diizzayy** [comment, marked as answer]
@Gyurmatag 

The code below should fit your needs.

`./nuxt-app/composables/myFetch.ts`
```ts
import type { FetchOptions } from 'ohmyfetch'

export const useCustomFetch = (path: string, opts?: FetchOptions) => {
  return $fetch(path, { baseURL: 'https://example.com', ...(opts && { ...opts }) })
}
```

**Reply by Gyurmatag** [reply]
@Diizzayy 
Yes, its' working. But I want to get my API_BASE URL from my .env file through this way:
```
const config = useRuntimeConfig();
const baseURL = config.API_BASE;
```
That gives me this error:
```
nuxt instance unavailable
  at useNuxtApp (file://./.nuxt/dist/server/server.mjs:337:13)  
  at Module.useRuntimeConfig (file://./.nuxt/dist/server/server.mjs:345:10)  
  at $id_604c20bc (file://./.nuxt/dist/server/server.mjs:3113:39)  
  at async __instantiateModule__ (file://./.nuxt/dist/server/server.mjs:15448:3)

```
Is there any other way to get the runTimeConfig properties in ./nuxt-app/composables/myFetch.ts ?

**Reply by Diizzayy** [reply]
Just ensure that you're setting the API_BASE in your runtime config as seen below:

`nuxt.config.ts`
```ts
import { defineNuxtConfig } from 'nuxt3'

export default defineNuxtConfig({
  publicRuntimeConfig: {
    API_BASE: '' // setting `API_BASE` in your `.env` will override this
  }
})
```

`./nuxt-app/composables/myFetch.ts`
```ts
import type { FetchOptions } from 'ohmyfetch'

export const useCustomFetch = (path: string, opts?: FetchOptions) => {
  const { API_BASE: baseURL } = useRuntimeConfig()

  if (!baseURL) throw new Error('baseURL is not defined')

  return $fetch(path, { baseURL, ...(opts && { ...opts }) })
}
```

@Gyurmatag  This code will work as you expect! Be sure to mark this discussion as answered if this helps.

**Reply by Gyurmatag** [reply]
Yeah , thank you. It's working. Just a little correction, it's working with useFetch(), like this:

```
export const useCustomFetch = (path: string, opts?: FetchOptions) => {
  const { API_BASE: baseURL } = useRuntimeConfig();
  return useFetch(path, { baseURL, ...(opts && { ...opts }) });
};
```

Am I right, @Diizzayy ?

**Reply by Diizzayy** [reply]
@Gyurmatag  👍🏽  indeed, that should work properly
